### PR TITLE
fix pep8

### PR DESCRIPTION
**Pull request by ocefpaf**
@wesleybowman I would wait for @efiring input on this small re-factor b/c he may be on the side of just ignoring the pep8 import failure and keep `ut_constants` in `__init__` instead.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,13 +11,13 @@ matrix:
...
     env: TEST_TARGET=default
-  - python: 3.5
-    env: TEST_TARGET=coding_standards
   - python: 3.6
     env: TEST_TARGET=default
-  allow_failures:
   - python: 3.5
     env: TEST_TARGET=coding_standards
+  allow_failures:
+  - python: 3.6
```
Comment (ocefpaf) [inline .travis.yml]: This does not mean much but I like to run the coding standards with the latest Python available.

File: setup.py [hunk 2]
```diff
@@ -1,7 +1,5 @@
 import os
-import sys
 from setuptools import setup
-from setuptools.command.test import test as TestCommand
```
Comment (ocefpaf) [inline setup.py]: These are left-overs from #51 and we should merge these changes even if the rest is not.

File: utide/__init__.py [hunk 3]
```diff
@@ -1,22 +1,5 @@
...
-# At least for now, use NaNs rather than masked arrays.
-ut_constants = loadbunch(_ut_constants_fname, masked=False)
-ut_constants = convert_unicode_arrays(ut_constants)
-
-constit_names = list(ut_constants.const.name)
-
-# Make a dictionary for index lookups.
-constit_index_dict = dict([(name, i) for (i, name) in
-                           enumerate(constit_names)])
-
```
Comment (ocefpaf) [inline utide/__init__.py]: Most of the changes are a consequence of this one. I created a new private module named `_ut_constants` just to carry the constants created here.

Comment (wesleybowman) [reply utide/__init__.py]: Looks a bit cleaner. I like it.

File: utide/_ut_constants.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+# At least for now, use NaNs rather than masked arrays.
+ut_constants = loadbunch(_ut_constants_fname, masked=False)
+ut_constants = convert_unicode_arrays(ut_constants)
+
+constit_names = list(ut_constants.const.name)
+
+# Make a dictionary for index lookups.
+constit_index_dict = dict([(name, i) for (i, name) in
+                           enumerate(constit_names)])
```
Comment (wesleybowman) [inline utide/_ut_constants.py]: We could also make this a class, could we not? 

```python
class Constants:
   def __init__(self):
        # load all the variables
        # do all the stuff
```

Then we would just import the class, and use it with `Constants.constit_index_dict`, or something like that.

Maybe we don't like that though, just throwing out an idea.

File: tests/test_solve.py [hunk 5]
```diff
@@ -119,10 +121,14 @@ def test_masked_input():
...
     phase_err = phase - elev_coef['g'][0]
     ts_recon = reconstruct(time, elev_coef).h
+    assert isinstance(ts_recon, np.ndarray)
 
     # pure smoke testing of reconstruct
     vel = reconstruct(time, speed_coef)
+    assert isinstance(vel, Bunch)
+
     elev = reconstruct(time, elev_coef)
+    assert isinstance(elev, Bunch)
```
Comment (wesleybowman) [inline tests/test_solve.py]: 👍 for all the new asserts

**Comment by efiring** [comment]
What is the pep8 failure that this is addressing?
A consequence of this refactor as it stands is that formerly public attributes become private.  I think they should be public, though; they are useful to the user.

File: utide/_ut_constants.py (hunk 4 above)
Comment (efiring) [reply utide/_ut_constants.py]: I don't think that using a class is appropriate here.  You would need to import it *and* create an instance; but that would have no advantage over the simpler structure in which the loading is done at package import time, and the constants are then accessed directly.

**Comment by ocefpaf** [comment]
> What is the pep8 failure that this is addressing?

Imports not at the top (`E402`).

> A consequence of this refactor as it stands is that formerly public attributes become private. I think they should be public, though; they are useful to the user.

We can make those public if you want to move this idea forward, let me update the PR...

File: utide/__init__.py [hunk 6]
```diff
@@ -1,27 +1,13 @@
...
 from ._solve import solve
 from ._reconstruct import reconstruct
+from ._ut_constants import ut_constants, constit_index_dict
 
 __all__ = ['solve',
-           'reconstruct']
+           'reconstruct',
+           'ut_constants',
+           'constit_index_dict']
```
Comment (ocefpaf) [inline utide/__init__.py]: @efiring if you don't like this solution I can revert everything, but the `setup.py` fixes, and get the 0.2.1 release out.

**Comment by ocefpaf** [comment]
> I think it is fine now.

Do you have merge rights?

**Comment by efiring** [comment]
No, I don't see the big green button.

**Comment by ocefpaf** [comment]
I do, but I wanted to avoid self-merges. I'll abuse my new powers for now :grimacing:

**Comment by wesleybowman** [comment]
Haha, self merge away

**Comment by wesleybowman** [comment]
It seems that right now @efiring doesn't have `Collaborator` status, hence why he couldn't merge.

@ocefpaf now that you are a collaborator, I don't think you have to fork anymore, you can just create new branches here (which means we could also do everything via the command line git, like merge, rebase, etc.)

**Comment by ocefpaf** [comment]
> I don't think you have to fork anymore

I subscribe to the fork-branch-PR model, here "fork" is the not the old bad sense of splitting the software, but rather a git/GitHub workflow.

For example, when forking a PR will trigger the CIs only once instead of twice. I do that even for my own projects.

**Pull request merged**